src/java/org/apache/cassandra/io/compress/CompressionMetadata.java

```diff
@@ -263,32 +327,28 @@ public Chunk chunkFor(long position)
     public long getTotalSizeForSections(Collection<SSTableReader.PartitionPositionBounds> sections)
     {
         long size = 0;
-        long lastOffset = -1;
         for (SSTableReader.PartitionPositionBounds section : sections)
         {
-            int startIndex = (int) (section.lowerPosition / parameters.chunkLength());
-
-            int endIndex = (int) (section.upperPosition / parameters.chunkLength());
-            if (section.upperPosition % parameters.chunkLength() == 0)
-                endIndex--;
+            int sectionStartIdx = chunkIndex(section.lowerPosition);
+            int sectionEndIdx = chunkIndex(section.upperPosition - 1); // we need to include the last byte of the seciont but not the upper position (which is excludded)
 
-            for (int i = startIndex; i <= endIndex; i++)
+            for (int idx = sectionStartIdx; idx <= sectionEndIdx; idx++)
             {
-                long offset = i * 8L;
-                long chunkOffset = chunkOffsets.getLong(offset);
-                if (chunkOffset > lastOffset)
-                {
-                    lastOffset = chunkOffset;
-                    long nextChunkOffset = offset + 8 == chunkOffsetsSize
-                                                   ? compressedFileLength
-                                                   : chunkOffsets.getLong(offset + 8);
-                    size += (nextChunkOffset - chunkOffset);
-                }
+                long chunkOffset = chunkOffset(idx);
+                long nextChunkOffset = nextChunkOffset(idx);
+                size += nextChunkOffset - chunkOffset;
             }
         }
         return size;
     }
 
+    private long nextChunkOffset(int relChunkIdx)
+    {
+        return relChunkIdx + 1 >= chunksCount
+               ? compressedFileLength
+               : Math.min(chunkOffset(relChunkIdx + 1), compressedFileLength);
```

Comment:
This `min` should not be necessary (also, the +1 and >= above cancel out, `relChunkIdx > chunkCount` has the same effect).

Reply:
indeed 🤦🏻